src/cdk/portal/dom-portal-outlet.ts

```diff
@@ -93,6 +93,28 @@ export class DomPortalOutlet extends BasePortalOutlet {
     return viewRef;
   }
 
+  /**
+   * Attaches a DOM portal by transferring its content into the outlet.
+   * @param portal Portal to be attached.
+   */
+  attachDomPortal(portal: DomPortal) {
+    // Note that we need to convert this into an array, because `childNodes`
+    // is a live collection which will be updated as we add/remove nodes.
+    let transferredNodes = Array.from(portal.element.childNodes);
```

Comment:
Can you think of any scenarios in which this would cause DOM nodes to be retained when they would otherwise be garbage collected? I'm wondering if we should say that the "destroy" behavior just destroys the nodes instead of returning them to their original location

Reply:
I don't think so. Assuming that this is the last place that they're being referenced, they should be collected once the dispose function is run and the `transferredNodes` is set to null.